src/legacy/core_plugins/kibana/public/visualize/wizard/search_selection/search_selection.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  EuiModalBody,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+  EuiTab,
+  EuiTabs,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React from 'react';
+
+import { VisType } from 'ui/vis';
+
+import { SavedObjectFinder } from 'ui/saved_objects/components/saved_object_finder';
+
+interface SearchSelectionProps {
+  onSearchSelected: (searchId: string, searchType: string) => void;
+  visType: VisType;
+}
+
+interface SearchSelectionState {
+  selectedTab: TabProps;
+}
+
+interface TabProps {
+  id: string;
+  name: string;
+  savedObjectFinder: React.ReactNode;
+}
+
+export class SearchSelection extends React.Component<SearchSelectionProps, SearchSelectionState> {
+  private tabs: TabProps[] = [];
+  private fixedPageSize: number = 8;
+
+  constructor(props: SearchSelectionProps) {
+    super(props);
+
+    this.tabs = [
```

Comment:
I know the add panel does this as well, but AFAIK and IMHO the SavedObject finder should always be initialized as part of the render function - if we are doing it in the constructor, the definitions can get stale (e.g. when a prop changes, it won't re-render correctly).

https://stackoverflow.com/questions/47875097/add-element-to-a-state-react

Reply:
Thank you for the comment. I updated `SearchSelection`.